addons/survey/models/survey_user_input.py

```diff
@@ -243,6 +244,13 @@ def _mark_done(self):
         Challenge = self.env['gamification.challenge'].sudo()
         badge_ids = []
         for user_input in self:
+            user_input.message_post(
+                author_id=self.env.ref('base.partner_root').id,
+                body=_('%(participant)s just participated in %(survey_title)s.',
+                       participant=user_input.partner_id.name or _("Someone"),
```

Comment:
`partner_id` is not required on this model. What will happen if `partner_id` is `False`? :eyes:

Reply:
Ok, I did a small test and `user_input.partner_id` will return a `<class 'odoo.api.res.partner'>` object and not a boolean. So, we can safely access the `name` attribute :smile: